### PR TITLE
Issue #SB-5245 fix: Handle browser back button.

### PR DESCRIPTION
**Pull request by vaishnavimanjunath**

### CODE REVIEW

File: src/app/client/src/app/modules/workspace/services/work-space/workspace.service.ts [hunk 1]
```diff
@@ -162,16 +162,20 @@ export class WorkSpaceService {
...
-  toggleWarning() {
+  toggleWarning(type) {
     this.showWarning = sessionStorage.getItem('inEditor');
     if (this.showWarning === 'true') {
-      this.listener = function (event) {
+      this.listener = (event) => {
         window.location.hash = 'no';
         if (event.state) {
-          alert(this.resourceService.messages.imsg.m0037);
+          if (type !== '' && type !== undefined) {
```
Comment (harishgilimi) [inline src/app/client/src/app/modules/workspace/services/work-space/workspace.service.ts]: **if (type !== '' && type !== undefined)** can be **if(!type)**

File: src/app/client/src/app/modules/workspace/components/create-editors/content-editor/content-editor.component.ts [hunk 2]
```diff
@@ -186,7 +186,7 @@ export class ContentEditorComponent implements OnInit, AfterViewInit, OnDestroy
       document.getElementById('contentEditor').remove();
     }
     sessionStorage.setItem('inEditor', 'false');
-    this.workspaceService.toggleWarning();
+    this.workspaceService.toggleWarning('');
```
Comment (harishgilimi) [inline src/app/client/src/app/modules/workspace/components/create-editors/content-editor/content-editor.component.ts]: no need to send '' as param

Comment (vaishnavimanjunath) [reply src/app/client/src/app/modules/workspace/components/create-editors/content-editor/content-editor.component.ts]: Then how should I send?

Comment (harishgilimi) [reply src/app/client/src/app/modules/workspace/components/create-editors/content-editor/content-editor.component.ts]: this.workspaceService.toggleWarning();

**Pull request merged**